### PR TITLE
Add more lint checks

### PR DESCRIPTION
**Pull request by dnephin**
These are split into separate commits so that you can see which changes were made for which linter. I can squash it before we merge if you want.

Adds 5 new linters and fixes some errors they uncovered.

`gocycle` has to ignore a few functions, and currently has an extremely high limit. I'd like to eventually get that down to much closer to 10, and work on fixing some of the whitelisted offenders.

### CODE REVIEW

File: cli/command/service/logs.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func runLogs(dockerCli *command.DockerCli, opts *logsOptions) error {
 		if !client.IsErrServiceNotFound(err) {
 			return err
 		}
-		task, _, err := cli.TaskInspectWithRaw(ctx, opts.target)
```
Comment (vdemeester) [inline cli/command/service/logs.go]: These `err` should be handled (can be in a different PR but..)
cc @dperny I can't find the `moby/moby` PR that does that.

File: cli/command/stack/ps.go [hunk 2]
```diff
@@ -72,5 +72,5 @@ func runPS(dockerCli *command.DockerCli, opts psOptions) error {
 		}
 	}
 
-	return task.Print(dockerCli, ctx, tasks, idresolver.New(client, opts.noResolve), !opts.noTrunc, opts.quiet, format)
+	return task.Print(ctx, dockerCli, tasks, idresolver.New(client, opts.noResolve), !opts.noTrunc, opts.quiet, format)
```
Comment (vdemeester) [inline cli/command/stack/ps.go]: 😍

File: cli/command/service/logs.go (hunk 1 above)
Comment (dnephin) [inline cli/command/service/logs.go]: ya, I had the same thought, but didn't want to do it here. I know I reviewed that PR, and I think I noticed it at the time, but convinced myself that the error wouldn't be an issue.

How about I open an issue for it to track it, and we merge this?

Comment (vdemeester) [inline cli/command/service/logs.go]: @dnephin sounds good to me 😉

Comment (dnephin) [inline cli/command/service/logs.go]: Cool , opened #22

Comment (dperny) [inline cli/command/service/logs.go]: Possibly this: https://github.com/moby/moby/pull/32876 ?

Comment (vdemeester) [inline cli/command/service/logs.go]: @dperny you are my hero 😻

Comment (dperny) [inline cli/command/service/logs.go]: @vdemeester http://1.bp.blogspot.com/-mlzxRZudNEs/U2DeUZgtzoI/AAAAAAAAZNs/nTNS8QoHac4/s1600/done.jpg

**Pull request merged**